### DISCUSSION TITLE
eventManager without react

### DISCUSSION THREAD

**Discussion by farfromrefug** [visgl/deck.gl#5646] [Q&A]
I am integrating deck.js inside [Nativescript](https://nativescript.org/). It is a cross platform mobile framework which binds the JS world with native (java, objc).
I got deck.js to render but i dont get any "gesture".
This is mostly due to the fact that i initiate `Deck` with the `gl` option and thus no DOM element to register events from.
In Nativescript we have all gestures, i would just need some guidance on how to "pass" Nativescript gesture events to deck.gl.
It seems i need to to that with `EventManager` but i could not find how to do that. Is there some example (like maybe mock test of gestures ?) that would help me in implementing this?

Thanks

**Comment by ibgreen** [comment]
Event handling is implemented by mjolnir.js. Perhaps you can implement a compatible EventManager that binds to your nativescript APIs.

**Reply by farfromrefug** [reply]
@ibgreen could you link me to where it is instantiated?

**Reply by ibgreen** [reply]
https://github.com/visgl/deck.gl/search?q=new+EventManager

**Reply by farfromrefug** [reply]
@ibgreen i have looked at the code and i think i found what s wrong.
Using deck without react the `EventManager` is created with `HammerManagerMock` which does not handle events.
I am good with that and i understand i need to implement my own `EventManager` class.
However `Deck` does not allow me (from what i have seen) to do that without trickery. 
I would need a `Deck` constructor option `eventManager` which would be used if exist instead of the eventManager from `mjolnir.js`
What do you think?

**Reply by ibgreen** [reply]
Event management works without React. mjolnir.js is not a React library. I am not quite sure I understand.

Regardless, I think adding an eventManager prop to deck.gl should be an acceptable change.

**Reply by farfromrefug** [reply]
@ibgreen sorry for not explaining correctly. 
I thought about the `Deck` constructor parameter. would be even better for it to by a "type" like we do for controller. that way the "default" `EventManager` would not even be constructed and thus I could with webpack "drop" mjolnir.js and hammer.js thus reducing my app size.
Right now I am using webpack to alias the `EventManager` to my own implementation( using native mobile gestures)
